fusor/model.py

```diff
@@ -125,62 +79,56 @@ def schema_extra(schema, _):
+                    'gene_id': 'hgnc:12012',
                     'type': 'GeneDescriptor',
                     'label': 'TPM3',
                 },
                 'component_genomic_region': {
                     'id': 'TPM3:exon1-exon8',
                     'type': 'LocationDescriptor',
-                    'value': {
+                    'location': {
                         'sequence_id': 'ga4gh:SQ.ijXOSP3XSsuLWZhXQ7_TJ5JXu4RJO6VT',  # noqa: E501
                         'type': 'SequenceLocation',
                         'interval': {
-                            'start': 154192135,
-                            'end': 154170399,
-                            'type': 'SimpleInterval'
+                            'start': {
+                                'type': 'Number',
+                                'value': 154192135,
+                            },
+                            'end': {
+                                'type': 'Number',
+                                'value': 154170399,
+                            },
+                            'type': 'SequenceInterval',
                         }
                     }
                 }
             }
 
 
-class SequenceDescriptor(BaseModel):
-    """Define VRSATILE Sequence Descriptor class."""
-
-    id: StrictStr
-    type = 'SequenceDescriptor'
-    value: Optional[SequenceState]
-    value_id: Optional[StrictStr]
-    residue_type = 'SO:0000348'
-
-    _validate_id = validator('id', allow_reuse=True)(check_curie)
-    _validate_value_id = validator('value_id', allow_reuse=True)(check_curie)
-
-    @root_validator(pre=True)
-    def check_value_or_value_id_present(cls, values):
-        """Check that at least one of {`value`, `value_id`} is provided."""
-        msg = 'Must give values for either `value`, `value_id`, or both'
-        value, value_id = values.get('value'), values.get('value_id')
-        assert value or value_id, msg
-        return values
-
-    @validator('value')
-    def check_dna_nucleobases(cls, v):
-        """Check that sequence consists of DNA nucleobases only"""
-        msg = 'Linker sequence must consist only of {A,C,G,T}'
-        assert set(v.sequence.upper()) <= set('ACGT'), msg
-        return v
-
-
 class LinkerComponent(BaseModel):
     """Define Linker class (linker sequence)"""
 
     component_type = 'linker_sequence'
     linker_sequence: SequenceDescriptor
 
+    @validator('linker_sequence')
+    def validate(cls, v):
+        """Enforce nucleotide base code requirements on sequence literals."""
+        if isinstance(v, dict):
+            try:
+                sequence = v['sequence']
+            except KeyError:
+                raise TypeError
+        elif isinstance(v, SequenceDescriptor):
+            sequence = v.sequence
+        else:
+            raise TypeError
+        msg = 'Linker sequence must consist only of {A,C,G,T}'
+        assert set('ACGT') <= set(sequence), msg
```

Comment:
Tests fail if sequence is `AAAAAAAA`. You should add a test to handle these kinds of cases (contains valid characters, but not all of the valid characters). 
```suggestion
        assert set('ACGT') >= set(sequence), msg
```

Reply:
brutal!!!